src/main/scala/corpus.scala

```diff
@@ -0,0 +1,87 @@
+package treesitter.lsp
+
+import parsley.character.*
+import parsley.debug.*
+import parsley.combinator.*
+import parsley.Parsley
+
+enum LispNode:
+  case Leaf(name: String)
+  case Nest(title: String, nodes: List[LispNode])
+
+object LispNode:
+  export parsers.parser
+
```

Comment:
I usually keep a separate `private object lexers` around, so that things that need to handle whitespace internally are kept away from the main parser.

```scala
private object lexer:
  val nodeName = ....wsp
  
  extension [A](p: Parsley[A]) def wsp ...
```

Reply:
Inside that, I'd add a `fully` combinator, which reads leading whitespace, a parser, then possibly `eof`.

Reply:
outside of this object, we shouldn't need to think about whitespace _at all_, so you'll also want to expose:

```scala
given Conversion[Char, Parsley[Char]] = c => char(c).wsp
// ditto for `String`
```
(or whatever the Scala3 syntax is).

That way, in the main parser, you can just use `'('` and `')'` and they handle whitespace correctly (and remove the noise of `char`)